src/components/generate.js

```diff
@@ -0,0 +1,51 @@
+import TemplateStyleSheet from '../lib/template-stylesheet.json'
+
+let GENERATED_FILE = null
+
+/**
+ * Reads the values from local storage
+ * Appends css attribute
+ * @returns array: data that can be converted to css file.
+ */
+const getData = () => {
+  const data = []
+  data.push( ':root {\n' )
+  data.push( '/* Generated using ShabadOS Theme Generator */\n' )
+  Object.keys( TemplateStyleSheet ).forEach( key => {
+    if ( localStorage[key] !== 'none' && localStorage[key] !== 'inherit' ) {
+      data.push( `${key}:${localStorage[key]};\n` )
+    }
+  } )
+  data.push( '}' )
+  return data
+}
+
+/**
+ * Converts the data[] into a blob for storage
+ * @returns blob URL
+ */
+const makeCssFile = () => {
+  const dataFile = new Blob( getData(), { type: 'text/plain' } )
```

Comment:
Why does new Blob not take a string? i'm sure it does, then, you can use `.join('\n')` or build a reducer

Reply:
It is possible to have a long string and then add then make a Blob but I think the array way is easier.